Stack trace local variables styling euitable

ffknob opened the pull request:
## Summary

Changed the 'local variables' style in Stacktrace to use Eui, and not anymore the inline styling. 
It looks a little different now, I choose to get rid of the "VariablesToggle" and use EuiAccordion instead.

Before:

<img width="1599" alt="local-variables-before" src="https://user-images.githubusercontent.com/23581855/66059539-90061300-e512-11e9-900c-e02823529297.png">


After:

![local-variables-after](https://user-images.githubusercontent.com/23581855/66059507-84b2e780-e512-11e9-8dc6-f6692a226db9.png)

ffknob committed:
changed the 'local variables' style in Stacktrace to use Eui

elasticmachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

elasticmachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

ffknob committed:
Changed the 'local variables' style in Stacktrace to use Eui, and not anymore the inline styling.
It looks a little different now, I choose to get rid of the "VariablesToggle" and use EuiAccordion instead.

sqren commented on x-pack/plugins/console_extensions/spec/overrides/security.put_user.json:
```diff
@@ -3,7 +3,7 @@
       "metadata": {},
       "password": "",
-      "fullname": "",
+      "full_name": "",
```
Why is this changed?

sqren commented on x-pack/plugins/console_extensions/spec/overrides/xpack.security.put_user.json:
```diff
@@ -3,7 +3,7 @@
       "metadata": {},
       "password": "",
-      "fullname": "",
+      "full_name": "",
```
Same with this.

ffknob replied:
I guess this comes from another PR I did a while ago: https://github.com/elastic/kibana/pull/37033

Sorry, I am new to this whole Git flow process... I'm not sure why this came together with the modifications I made for this particular issue.

sqren replied:
No worries :)

sqren commented on x-pack/plugins/apm/public/components/shared/Stacktrace/Variables.tsx:
```diff
@@ -4,72 +4,41 @@
+        <EuiSpacer />
+        <EuiAccordion
+          id="local-variables"
+          buttonContent={this.localVariablesToogleButtonLabel()}
```
This doesn't need to be a separate function
```suggestion
          buttonContent={i18n.translate('xpack.apm.stacktraceTab.localVariablesToogleButtonLabel', { defaultMessage: 'Local variables' }) }
```

sorenlouv reviewed (approved):
Thanks for the contribution. Code changes LGTM. Can you add a before / after screenshot to the description?

@formgeist what do you think of the new look?

sqren commented:
@ffknob Will you remove the unrelated files, and then this is good to go.

formgeist commented:
@ffknob Thanks for taking this on 👍 I have some minor visual feedback on how the section is presented in the stackframe panel.

Would it be possible to move the accordion collapsed/expand button outside the stackframe background? And could we add some margins around the table itself too? Here's a screen that kind of shows what I'm looking for;

![apm-stackframe-local-vars-collapsed](https://user-images.githubusercontent.com/4104278/66385107-3a24e580-e9c0-11e9-927d-5900fb3aae12.png)
![apm-stackframe-local-vars-expaned](https://user-images.githubusercontent.com/4104278/66385109-3a24e580-e9c0-11e9-9e30-ccc09b1c6148.png)

ffknob committed:
removed function that generated the 'Local variables' label

ffknob committed:
removed the funciton that returned the  'Local variables' label

ffknob committed:
brought back some of the styling code so it looks more like how it used
to be

ffknob commented:
Hi guys, here's what've done...

1. 'Local variables' label defined by inline code, instead of a function
2. I *think* I managed to bring the other two unrelated files back to the originals (I just brought the content back to what it looks like in master... altought I still am not sure how they got into my commit... and also how is this change not part of master, since I think this PR I did a while ago was merged... anyway...)
3. Gave a shot to bring the look&feel more to what @formgeist asked for (yeah, that was bothering me too). I now use a EuHorizontalRule instead of EuiSpacer, but I had to bring back some of the old inline css code (and its imports)... Collapsed/Expanded image below

Collapsed
![collapsed](https://user-images.githubusercontent.com/23581855/66393254-6f253e00-e9a8-11e9-8c2a-ec6a2606c259.png)


Expanded
![expanded](https://user-images.githubusercontent.com/23581855/66393261-73515b80-e9a8-11e9-8fc9-7030bb50deb1.png)

ffknob committed:
applied euiAccordion class to the component

formgeist commented on x-pack/plugins/apm/public/components/shared/Stacktrace/Variables.tsx:
```diff
@@ -27,49 +27,34 @@ const VariablesContainer = styled.div`
   font-family: ${fontFamily};
```
This should be removed as it's causing the font to render as Open Sans instead of the default Kibana font, Inter UI.

ffknob replied:
That's right. Font-family removed.

formgeist reviewed (changes requested):
Just a minor nit on the container adding its own font-family.

formgeist commented:
> But I don't know why the link color did not get to "#006BB4" (@elastic/eui/dist/eui_theme_light.css:338)

I think it's because the default styling of the accordion toggle button is not the Kibana blue but instead the default text colour. I don't think that's too critical in this case. We can probably find a way to duplicate the button style from the library frames toggle.

<img width="235" alt="Screenshot 2019-10-08 at 14 55 48" src="https://user-images.githubusercontent.com/4104278/66397268-bb3da600-e9db-11e9-95b3-80d9d7fbbb23.png">

formgeist commented on x-pack/plugins/apm/public/components/shared/Stacktrace/Variables.tsx:
```diff
@@ -27,49 +27,34 @@ const VariablesContainer = styled.div`
-        )}
-      </VariablesContainer>
+      <React.Fragment>
+        <EuiHorizontalRule margin="none" />
```
I don't see this making any difference except it adds another border to the panel. I reckon we can remove the `EuiHorizontalRule` here.

ffknob replied:
Horizontal Rule removed

formgeist reviewed (changes requested):
Just another visual nit 🙂

ffknob committed:
removed font-family and rule component

ffknob commented:
Just removed both font-family and horizontal rule.

sqren commented:
@elasticmachine merge upstream

elasticmachine commented:
merge conflict between base and head

sqren commented:
@ffknob Can you pull in master and fix the conflict?

ffknob commented:
Wow! Now I think I really messed up... 
I tried to rebase it to master (CONTRIBUTING), I'm not even sure that's what I should've done...
How can I fix this? (or can't I?)

Here's the commands I used (I'd appreciate if you could point out where did I messed things up)

(by the way sorry for that... I'm only used to work on 'master', yeah, I know, I know)

git checkout master
git rebase upstream/master
git checkout stack-trace-local-variables-styling-euitable
git rebase master
yarn kbn bootstrap
node scripts/precommit_hook --fix
ERROR Filenames MUST use snake_case.
       - x-pack/plugins/reporting/.chromium/chromium-2fac04a-darwin.zip
       - x-pack/plugins/reporting/.chromium/chromium-2fac04a-linux.zip
       - x-pack/plugins/reporting/.chromium/chromium-2fac04a-windows.zip

git commit . --no-verify
[stack-trace-local-variables-styling-euitable 00515093a3] rebase to master
 3 files changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 x-pack/plugins/reporting/.chromium/chromium-2fac04a-darwin.zip
 create mode 100644 x-pack/plugins/reporting/.chromium/chromium-2fac04a-linux.zip
 create mode 100644 x-pack/plugins/reporting/.chromium/chromium-2fac04a-windows.zip

git pull
git commit
git add .
git commit
git push origin stack-trace-local-variables-styling-euitable

ffknob commented:
I found an article with instructions and then did a local hard reset  last commit before breaking everything)

git reset --hard 2a9141a3e7b1e9695bea94f538acb779085d4bea

It then says to --force push... Should I?
This time I prefer to check with you guys for instructions... All I want is to go back to my only 1 file commit

Once again sorry for the git-noobness and thanks for the help

sqren commented:
Hey @ffknob 

no worries, that happens.  Without looking more into this I don't know what the best approach to revert is. You are welcome to try force pushing 👍

ffknob committed:
back to full_name

ffknob commented:
> Hey @ffknob
> 
> no worries, that happens. Without looking more into this I don't know what the best approach to revert is. You are welcome to try force pushing

Back to 1 file changed.
Now I'll do what I should've done in the first place:

> Can you pull in master and fix the conflict?

How can I verify this conflict?
I guess it will show as a result of the "kibana-ci Expected — Waiting for status to be reported", right?

And now that I managed to bring back to my 1 file changed commit, could you please guide me with the "pull in master" part? I guess not the fetch/rebase/pull (some of that I took from the CONTRIBUTING guide)

sqren commented:
retest

sqren commented:
> How can I verify this conflict?

Looks like you fixed the conflict already. 👍 I've kicked off a build and will merge if it's successful. Thanks for the contribution

ffknob committed:
removed unused components

ffknob commented:
>      9 import { EuiHorizontalRule, EuiAccordion } from '@elastic/eui';
>                 ~~~~~~~~~~~~~~~~~
>      x-pack/legacy/plugins/apm/public/components/shared/Stacktrace/Variables.tsx:14:3 - error TS6133: 'fontFamily' is declared but its value is never read.
>      14   fontFamily,

Just removed them and pushed back into the PR.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/3242/)
* Commit: adce9b05d061640874622f51ae361a31efe763ed

jbudz commented:
Dropped a few labels, we'll let the apm team take this :).  Thanks for the PR!

sqren commented:
retest

sqren commented:
@elasticmachine merge upstream

elasticmachine commented:
head repository does not exist

sqren commented:
@ffknob Can you please merge in the latest master? Then I'll merge this.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/4561/)
* Commit: 043fa4def6a729b3b3451937509f45e324815409

ffknob commented:
Hi @sqren 
Coul you please verify this new PR I have created for this?
https://github.com/elastic/kibana/pull/48208

I'll explain:

- Yesterday I had a problem and Ithought it would help to delete my Kibana fork
- The thing is I didn't realized the links between open PRs and my repo would be forever lost
- Then I was not able to rebase the branch that generated this PR to master anymore
- So I created a new PR (in my new Kibana fork) with exactly the same change
- In that PR I referenced this one here

So perhaps that one is the one to be merged and whis one should be closed...
The good thing is I think that one is already up to master.

sqren commented:
Closed in favor of #48208
